lib/rex/proto/mysql/client.rb

```diff
@@ -28,6 +28,28 @@ def current_database
           # Current database is stored as an array under the type 1 key.
           session_track.fetch(1, ['']).first
         end
+
+        # @return [Hash] A hash with :arch and :version keys if the query was successful, otherwise empty.
+        def query_server_version_vars
```

Comment:
I was wondering what your thoughts would be on maybe expanding this new method.

Reason I'm thinking this might be a good idea is that `auxiliary/admin/mysql/mysql_enum.rb` makes use of something similar:
https://github.com/rapid7/metasploit-framework/blob/3c8f43e23ef4a95955f557c5422ceea605c1436a/modules/auxiliary/admin/mysql/mysql_enum.rb#L69-L83

It might be nice to update your new module to handle all the variable so that `mysql_enum` can also make use of it.